src/app/modules/project/edit/edit.component.ts

```diff
@@ -130,23 +138,40 @@ export class EditComponent implements OnInit {
     const edittedProject: ProjectUpdate = this.editProjectForm.value;
     edittedProject.collaborators = this.collaborators;
 
-    this.projectService
-      .put(this.project.id, edittedProject)
-      .pipe(
-        finalize(() => {
-          this.submitEnabled = false;
-        })
-      )
-      .subscribe(() => {
+    this.fileUploader.uploadFiles().subscribe(uploadedFiles => {
+      if (uploadedFiles[0]) {
+        edittedProject.fileId = uploadedFiles[0].id;
+        this.editProject(edittedProject);
+      } else {
```

Comment:
You should check if there is a file before uploading it. 

editProject should be called after uploading the file (if that is the case).

Reply:
That should be checked indeed, I don't think that that's the problem tho.
fileUploader appears to be undefined because it's not rendered instantly because everything is in `<ng-container>` & `<ng-temaplate>`

Reply:
if uploadFiles() does not return anything if the file wasn't uploaded if that's what you mean